### PR TITLE
Issue 703 -- support for Decimal type

### PR DESCRIPTION
**Pull request by picturedots**
https://github.com/graphql-python/graphene/issues/703

### CODE REVIEW

**Comment by jkimbo** [comment]
@picturedots you'll need to setup a virtualenv with python 2.7 Then run the commands in the contributing section in the readme: https://github.com/graphql-python/graphene#contributing

Looks like the flake8 (for linting) is not included in the requirements so you'll need to install it first before running it.

**Comment by jkimbo** [comment]
Also thanks for the PR! It's looking great

**Comment by jplock** [comment]
Anyway we can get this into the next release?

**Comment by picturedots** [comment]
I got stalled on the tests and forgot to come back to this -- when is the next release?

**Comment by jplock** [comment]
I don't think there's any set schedule, I've just been noticing some release-related activity on the various graphql-python projects and thought a new graphene release was imminent.

**Comment by jplock** [comment]
@picturedots maybe try adding `Decimal` into `__all__` in `graphene\__init__.py` to fix the linting issue

**Comment by picturedots** [comment]
I appreciate the updates to the "Contributing" section in README.MD -- I will see if I can finally get this finished this week.

**Comment by jplock** [comment]
To be ahead of https://github.com/graphql-python/graphene/pull/794, we could raise assertion errors.

**Comment by picturedots** [comment]
@jplock It looks like #794 is going to provide a helper method once its accepted, so maybe it's best to wait.  I did, however fix some linting errors  that complained about `assert x == None` and converted to `assert is None`.

**Comment by picturedots** [comment]
It looks like there are black formatting errors in `test_mutation.py` from the master branch that is breaking tox for me.  @sebdiem  do you know anything about this?  If somebody can fix it on master then I can merge in that change to clear the Travis errors.

**Comment by sebdiem** [comment]
@picturedots done: https://github.com/graphql-python/graphene/pull/802

**Comment by picturedots** [comment]
This should be all good to go now, @jplock  -- unfortunately it looks like I missed the latest release.

**Comment by jplock** [comment]
Thanks for all the efforts @picturedots!  @jkimbo @syrusakbary think this could make the next release?

File: graphene/types/decimal.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    @classmethod
+    def parse_literal(cls, node):
+        if isinstance(node, ast.StringValue):
+            return cls.parse_value(node.value)
+
+    @staticmethod
+    def parse_value(value):
+        try:
+            return _Decimal(value)
+        except ValueError:
```
Comment (maarcingebala) [inline graphene/types/decimal.py]: Shouldn't it check for `decimal.DecimalException` here, as for example [Django does](https://github.com/django/django/blob/master/django/forms/fields.py#L346)? I've copied your changes to my repo (as it hasn't been released yet) and I'm trying to use it in my mutations, but when providing invalid values like an empty string I'm getting:
```
{
  "errors": [
    {
      "message": "[<class 'decimal.ConversionSyntax'>]"
    }
  ]
}
```

Comment (jplock) [reply graphene/types/decimal.py]: @maarcingebala can you submit a pull request with this change so it can be reviewed?

**Pull request merged**